src/components/AnnotationLineConnector/AnnotationLineConnector.js

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useContext, useState } from 'react';
+import NoteContext from 'components/Note/Context';
+import { useSelector, shallowEqual } from 'react-redux';
+import core from 'core';
+import selectors from 'selectors';
+import { createPortal } from 'react-dom';
+
+import './AnnotationLineConnector.scss';
+
+
+const LineConnectorPortal = ({ children }) => {
+  const mount = document.getElementById("line-connector-root");
+  const el = document.createElement("div");
+  el.setAttribute('data-element', 'annotationLineConnector');
+
+  useEffect(() => {
+    mount.appendChild(el);
+    return () => mount.removeChild(el);
+  }, [el, mount]);
+
+  return createPortal(children, el);
+};
+
+
+const AnnotationLineConnector = ({ annotation, noteContainerRef }) => {
+  const [notePanelWidth, isLineOpen, isNotesPanelOpen] = useSelector(
+    state => [
+      selectors.getNotesPanelWidth(state),
+      selectors.isElementOpen(state, 'annotationLineConnector'),
+      selectors.isElementOpen(state, 'notesPanel'),
+    ],
+    shallowEqual,
+  );
+
+  //Horizontal Line One
+  const [horizontalLineOneWidth, setHorizontalLineOneWidth] = useState(0);
+  const [horizontalLineOneTop, setHorizontalLineOneTop] = useState(0);
+  const [horizontalLineOneRight, setHorizontalLineOneRight] = useState(0);
+
+  //Horizontal Line Two
+  const [horizontalLineTwoWidth, setHorizontalLineTwoWidth] = useState(0);
+  const [horizontalLineTwoTop, setHorizontalLineTwoTop] = useState(0);
+  const [horizontalLineTwoRight, setHorizontalLineTwoRight] = useState(0);
+
+  const { isSelected } = useContext(NoteContext);
+  const displayMode = core.getDisplayModeObject();
+  const annotationWindowCoords  = displayMode.pageToWindow({ x: annotation.getX(), y: annotation.getY() }, annotation.PageNumber);
+
+  useEffect(() => {
+    const { x: annotOriginX, y: annotOriginY } = annotationWindowCoords;
+    const { scrollTop } = core.getScrollViewElement();
+    const notePanelLeftPadding = 16;
+
+    // Annot Width to pixels
+    const annotationRightEdgeX = annotation.getX() + annotation.getWidth();
```

Comment:
If you want I think you could take advantage of the `getAnnotationPosition` function `getPopupPosition.js` to clean up the code here a bit(just for removing duplicates). I think that function calculates the rect of an annotation in window coordinates.